### PR TITLE
[WIP] Add label-free qleace

### PR DESCRIPTION
**Pull request by luciaquirke**

### CODE REVIEW

File: concept_erasure/alf_qleace.py [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+        # Compute QLEACE component
+        # Compute the (covariance - mean covariance) matrix difference for each class
+        self.sigma_xx_z_.shape
+        mean_sigma_xx_z = self.sigma_xx_z_.mean(dim=0)
+        sigma_xx_z_diffs = self.sigma_xx_z_ - mean_sigma_xx_z
+
+        # Find the class that has the difference with the largest singular
+        # value (spectral norm)
+        svds: list[tuple[Tensor, Tensor, Tensor]] = [
+            torch.svd_lowrank(sigma_xx_z_diffs[i], q=1) for i in range(self.z_dim)
```
Comment (luciaquirke) [inline concept_erasure/alf_qleace.py]: increase svd iterations to 10

File: concept_erasure/alf_qleace.py [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+        return self
+
+    @cached_property
+    def eraser(self) -> AlfQLeaceEraser:
+        """Erasure function lazily computed given the current statistics."""
+        eye = torch.eye(
+            self.x_dim, device=self.global_mean_x.device, dtype=self.global_mean_x.dtype
+        )
+
+        # Compute QLEACE component
```
Comment (luciaquirke) [inline concept_erasure/alf_qleace.py]: Compute ALF-Q components

File: concept_erasure/alf_qleace.py [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+        # value (spectral norm)
+        svds: list[tuple[Tensor, Tensor, Tensor]] = [
+            torch.svd_lowrank(sigma_xx_z_diffs[i], q=1) for i in range(self.z_dim)
+        ]
+        spectral_norms = torch.stack([svd[1][0] for svd in svds])
+        z_idx = spectral_norms.argmax()
+
+        # Select the principal direction associated with the singular value
+        U, S, Vh = svds[z_idx]
+        principal_direction = U[:, 0]
```
Comment (luciaquirke) [inline concept_erasure/alf_qleace.py]: Use squeeze()

File: concept_erasure/alf_qleace.py [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+        self.sigma_xx_z_.shape
+        mean_sigma_xx_z = self.sigma_xx_z_.mean(dim=0)
+        sigma_xx_z_diffs = self.sigma_xx_z_ - mean_sigma_xx_z
+
+        # Find the class that has the difference with the largest singular
+        # value (spectral norm)
+        svds: list[tuple[Tensor, Tensor, Tensor]] = [
+            torch.svd_lowrank(sigma_xx_z_diffs[i], q=1) for i in range(self.z_dim)
+        ]
+        spectral_norms = torch.stack([svd[1][0] for svd in svds])
```
Comment (luciaquirke) [inline concept_erasure/alf_qleace.py]: Use squeeze()
